cli/command/task_test.go

```diff
@@ -0,0 +1,313 @@
+
+	c := NewContext(t, Args{"tsk_name"}, nil)
+	if err := taskCommand.read(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestReadTask_userInputErrors(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	taskCommand := NewTaskCommand(base.Command())
+
+	contexts := map[string]*cli.Context{
+		"Missing TASK_NAME arg": NewContext(t, nil, nil),
+	}
+
+	for name, c := range contexts {
+		if err := taskCommand.read(c); err == nil {
+			t.Fatalf("%s: error was nil!", name)
+		}
+	}
+}
+
+func TestReadTask_expiredTasks(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	taskCommand := NewTaskCommand(base.Command())
+
+	result := []*models.TaskSummary{
+		{TaskID: "tsk_id1"},
+	}
+
+	base.Resolver.EXPECT().
+		Resolve("task", "tsk_name").
+		Return([]string{"tsk_id1", "expired1", "expired2"}, nil)
+
+	base.Client.EXPECT().
+		ListTasks().
+		Return(result, nil)
+
+	base.Client.EXPECT().
+		ReadTask("tsk_id1").
+		Return(&models.Task{}, nil)
+
+	c := NewContext(t, Args{"tsk_name"}, nil)
+	if err := taskCommand.read(c); err != nil {
+		t.Fatal(err)
+	}
+
+}
+
+func TestListTasks(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	taskCommand := NewTaskCommand(base.Command())
+
+	base.Client.EXPECT().
+		ListTasks().
+		Return([]*models.TaskSummary{}, nil)
+
+	c := NewContext(t, nil, nil)
+	if err := taskCommand.list(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestReadTaskLogs(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	taskCommand := NewTaskCommand(base.Command())
+
+	base.Resolver.EXPECT().
+		Resolve("task", "tsk_name").
+		Return([]string{"tsk_id"}, nil)
+
+	query := buildLogQueryHelper("tsk_id", "start", "end", 100)
+
+	base.Client.EXPECT().
+		ReadTaskLogs("tsk_id", query).
+		Return([]*models.LogFile{}, nil)
+
+	flags := Flags{
+		"tail":  100,
+		"start": "start",
+		"end":   "end",
+	}
+
+	c := NewContext(t, Args{"tsk_name"}, flags)
+	if err := taskCommand.logs(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestReadTaskLogs_userInputErrors(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	taskCommand := NewTaskCommand(base.Command())
+
+	contexts := map[string]*cli.Context{
+		"Missing TASK_NAME arg": NewContext(t, nil, nil),
+	}
+
+	for name, c := range contexts {
+		if err := taskCommand.logs(c); err == nil {
+			t.Fatalf("%s: error was nil!", name)
+		}
+	}
+}
+
+func TestParseOverrides(t *testing.T) {
+	input := []string{
+		"container1:key1=val1",
+		"container1:key2=val2",
+		"container2:k1=v1"}
+
+	expected := []models.ContainerOverride{
+		{
+			ContainerName:        "container1",
+			EnvironmentOverrides: map[string]string{"key1": "val1", "key2": "val2"},
+		},
+		{
+			ContainerName:        "container2",
+			EnvironmentOverrides: map[string]string{"k1": "v1"},
+		},
+	}
+
+	output, err := parseOverrides(input)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, len(output), len(expected))
```

Comment:
Can't you just use `assert.Equal(t, expected, output)`?

Reply:
Yes you can.